packages/doenetml-worker/lib-doenetml-core/src/dast/ref_resolve.rs

```diff
@@ -461,6 +498,29 @@ impl Resolver {
     }
 }
 
+/// Get the name of `element` from its `name` attribute,
+/// where the `name` attribute must have exactly one text child to be considered valid.
+fn get_element_name(element: &FlatElement) -> Option<String> {
+    // XXX: we need to enforce restrictions on valid names. Is this the right place?
+
+    let name = element
+        .attributes
+        .iter()
+        .find(|attr| {
+            attr.name == "name"
+                && attr.children.len() == 1
+                && matches!(attr.children[0], UntaggedContent::Text(_))
+        })
+        .and_then(|attr| {
+            match (attr.children.len(), attr.children.first()) {
+                // A name attribute should have exactly one text child. Otherwise it is considered invalid.
+                (1, Some(UntaggedContent::Text(name))) => Some(name.clone()),
```

Comment:
Why are you doing these tests twice? It seems that a consequence is that if multiple `name=".."` attributes are specified, and one is valid, that one will be used. But if we assume there is only one `name` attribute, you can get rid of the `&&` clauses in the preceding `.find`

Reply:
Yes, it looks like we were trying to address the case we have multiple name attributes and we want to grab the valid one. But, I'm fine with just grabbing the first name attribute and using it only if it is valid.